### PR TITLE
add lists, link, more code block options; improve level

### PR DESCRIPTION
**Pull request by ghouston**
Markdown
* add `heading` or `header` alternatives for `level`
* fix `level` to edit at line start
* add `list` commands
* add `link` command
* add more code block languages: `ruby`, `shell`, `bash`, `json`

### CODE REVIEW

**Review by knausj85 — APPROVED** [review]
This looks good to me; I'm going to leave it open for a bit to give collaborators a chance to review the alternatives added

**Review by splondike — APPROVED** [review]
Tests out OK, merging.

File: lang/markdown/markdown.talon [hunk 1]
```diff
@@ -1,14 +1,49 @@
...
 
 {user.markdown_code_block_language} block:
     "```{markdown_code_block_language}"
     key(enter:2)
     "```"
     key(up)
+
+link:
+    "[]()"
+    key(left:3)
```
Comment (auscompgeek) [inline lang/markdown/markdown.talon:49]: I think we prefer `user.insert_between` for this sort of thing, right?

Comment (splondike) [reply lang/markdown/markdown.talon:49]: Yeah we probably do.

**Pull request merged**